Code/GraphMol/FileParsers/testMolSupplier.cpp

```diff
@@ -2835,6 +2830,69 @@ void testGitHub3517() {
   TEST_ASSERT(!sdsup.atEnd());
 }
 
+void testParsingInvalidChiralityLabels() {
+#ifdef RDK_BUILD_MAEPARSER_SUPPORT
+  constexpr const char *maeblock_template = R"DATA(f_m_ct {
+  i_m_ct_stereo_status
+  s_m_title
+  s_st_Chirality_1
+  :::
+  1
+  ""
+  %s
+  m_atom[5] {
+    # First column is Index #
+    r_m_x_coord
+    r_m_y_coord
+    r_m_z_coord
+    i_m_atomic_number
+    i_m_formal_charge
+    :::
+    1 -1.000000 -0.060606 0.000000 6 0
+    2 -1.000000 -1.560606 0.000000 6 0
+    3 -0.250000 -2.859644 0.000000 17 0
+    4 -2.500000 -1.560606 0.000000 9 0
+    5 0.299038 -0.810606 0.000000 35 0
+    :::
+  }
+  m_bond[4] {
+    # First column is Index #
+    i_m_from
+    i_m_order
+    i_m_to
+    :::
+    1 1 1 2
+    2 2 1 3
+    3 2 1 4
+    4 2 1 5
+    :::
+  }
+}
+)DATA";
+  for (const auto &invalid_chirality_label : {
+           "12_R_1_3_4_5",     // missing chiral atom
+           "12_ANR_1_3_4_15",  // missing substituent atom
```

Comment:
This looks like it should use catch2 generators:
https://github.com/catchorg/Catch2/blob/devel/docs/generators.md

One disadvantage of loop based tests in C++ is that you can't usually see which test conditions caused a test failure in the test output. Using the test framework's native test generator tool helps with this (as I know you know from your work with boost::test :) ).

Reply:
I was not aware of this `catch2` functionality. thanks!